### DISCUSSION TITLE
How can I wrap a MUI google maps autocomplete component in a controller?

### DISCUSSION THREAD

**Discussion by trevor-ofarrell** [react-hook-form/react-hook-form#2124] [Q&A]
I have followed this code sandbox featuring a autocomplete component for all countries: https://codesandbox.io/s/react-hook-form-controller-079xx?file=/src/MuiAutoComplete.js But it didn't work when trying to implement the shown way using this autocomplete component: https://material-ui.com/components/autocomplete/#GoogleMaps.tsx
What would be the proper way to wrap [this](https://material-ui.com/components/autocomplete/#GoogleMaps.tsx) with a controller to be used in a form in a another file? Currently I'm getting this error in the browser console when filling out the form: `Material-UI: The value provided to Autocomplete is invalid.
None of the options match with `{"description":"R4 Solutions Inc, Jarvis Avenue, Newark, CA, USA","id":"c6ea5c3dde5eb6884a7636c9fbd8df5d64a2d68d","matched_substrings":[{"length":2,"offset":0}],"place_id":"ChIJD40Shk--j4ARVdpeyi3MIOo","reference":"ChIJD40Shk--j4ARVdpeyi3MIOo","structured_formatting":{"main_text":"R4 Solutions Inc","main_text_matched_substrings":[{"length":2,"offset":0}],"secondary_text":"Jarvis Avenue, Newark, CA, USA"},"terms":[{"offset":0,"value":"R4 Solutions Inc"},{"offset":18,"value":"Jarvis Avenue"},{"offset":33,"value":"Newark"},{"offset":41,"value":"CA"},{"offset":45,"value":"USA"}],"types":["point_of_interest","establishment"]}`.
You can use the `getOptionSelected` prop to customize the equality test.`

**Comment by bluebill1049** [comment]
please provide a CSB, so we can help.

**Reply by trevor-ofarrell** [reply]
Here https://codesandbox.io/s/priceless-moon-oyrub?file=/GoogleMaps.js I don't own the account holding this api key and I can only use it from my Ip, so I couldn't get the results to appear in the code sandbox because it's presumably ran on a remote server, preventing me from recreating the issue, however if you have a Google maps API key maybe you could try it out. Thanks.
